molecularnodes/entities/molecule/base.py

```diff
@@ -321,3 +321,50 @@ def __repr__(self) -> str:
             The string representation of the Molecule object.
         """
         return f"<Molecule object: {self.name}>"
+
+
+class MoleculeSelector(Selector):
```

Comment:
In general, I would recommend fewer classes. E.g. just use the Selector class and remove one layer of indirection.  If there were additional inheritors of Selector I think it might make sense. So I would personally vote for adding the store_names_attribute to the selector class and ditching this Class. Alternatively: rename Selector  to `MoleculeSelector`.

But in the end its style not substance. So non blocking.

Reply:
I split it up so one could be completely 'anonymous' and work separate to any molecule, and then the `MoleculeSelector` is the one that is directly associated with the class. I could combine them though like you said.